components/script/dom/file.rs

```diff
@@ -45,6 +47,19 @@ impl File {
                            FileBinding::Wrap)
     }
 
+    // Construct from selected file message from file manager thread
+    pub fn new_from_selected(window: &Window, selected: SelectedFile) -> Root<File> {
+        let name = DOMString::from(selected.filename.to_str().expect("File name encoding error"));
+
+        // FIXME: fix this after PR #11221 is landed
+        let id = selected.id;
+        let slice = DataSlice::empty();
+
+        let global = GlobalRef::Window(window);
+        // XXX: as i64? looks like DOM codegen and std::fs don't agree on this type
```

Comment:
Seems like it won't matter since `Date` breaks down at i64::max_value anyway. If the cast breaks anything, it will be a billion years from now :p